### ISSUE TITLE
Creating new local branch triggers updating submodules?

### ISSUE THREAD

**Issue by SirIntruder** [gitextensions/gitextensions#3899]
I create new local branch, git extensions triggers submodule update.
I have "don't ask" set for submodule update, because I want that to happen every time I switch commits, but why should this be triggered for action which keeps head on the same commit?

Same goes for checking out between branches which point on the same single commit.

**Environment you encounter the issue:**

- GitExtensions version: 2.50.1
- GIT version: 2.14
- OS version: win10

**Comment by gerhardol** [comment]
Have you enabled Advanced-Confirmations-Update submodules on commit?
When you create a new branch, it checkouts the the branch which triggers the update.
There is no way to distinguish between checkouts here.

It may be possible to use GitEx scripts instead. (You can enable the script to run at pull instead.)

Another area where handling of submodules can be improved in Git (not GitEx issue). I still prefer submodules.

**Comment by SirIntruder** [comment]
Yes, submodule updates are happening because "Advanced-Confirmations-Update submodules on checkout" is on (what I called "don't ask"). In the end I turned it off, it is easier to update manually when needed than wait like half a minute on every action (that's how long it takes in our main repository with ~15 submodules). And yes, I am aware I can have dialog to ask me each time, but that's just annoying

Anyway, this is general issue with checking out, not just creating new branch. If head is on commit X, and I checkout branch which is also pointing to the same commit X, there is never need to update submodules, and GitExtensions should not update them no matter what value "Update submodules on checkout" is set.
